File: Code.java
//
// Definition for binary tree:
// class Tree<T> {
//   Tree(T x) {
//     value = x;
//   }
//   T value;
//   Tree<T> left;
//   Tree<T> right;
// }

boolean isTreeSymmetric(Tree<Integer> t) {
    if(t == null)
        return true;
    
    return symmetricCheck(t.left, t.right);
}

boolean symmetricCheck(Tree<Integer> left, Tree<Integer> right){
    
    if(left == null && right == null)
        return true;
    
    if(left != null && right != null && left.value.equals(right.value))
    {
        return symmetricCheck(left.left, right.right) && symmetricCheck(left.right, right.left);
    }
    
    return false;
}

File: Code.c
//
//
// Definition for binary tree:
// typedef struct tree_##name {
//   type value;
//   struct tree_##name *left;
//   struct tree_##name *right;
// } tree_##name;
//
// tree_##name *alloc_tree_##name(type v) {
//   tree_##name *t = calloc(1, sizeof(tree_##name));
//   t->value = v;
//   return t;
// }
// 

bool symCheck(tree_integer * tl, tree_integer * tr){
    if(tl == NULL && tr == NULL)
        return true;
    
    if(tl != NULL && tr != NULL && tl->value == tr->value)
        return symCheck(tl->left, tr->right) && symCheck(tl->right, tr->left);
    
    return false;
}

bool isTreeSymmetric(tree_integer * t) {
    if(t == NULL)
        return true;
    
    bool res = symCheck(t->left, t->right);
    
    return res;
}
